internal/controllers/machinedeployment/mdutil/util_test.go

```diff
@@ -30,12 +30,16 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apiserver/pkg/storage/names"
-	"k8s.io/klog/v2/klogr"
 	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+var (
+	ctx = ctrl.SetupSignalHandler()
```

Comment:
Note: we won't have a logger in this context, right? (Propably similar for other places, but I only see tests affected (if even))

Reply:
There is no logger, but ctrl.LoggerFrom has a fallback for this case. Although the fallback isn't logging anything if there is no global logger set 

See: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/log/log.go#L91-L99
(NullLogSink)

Reply:
Fine for me 👍